src/LinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -245,85 +245,88 @@ void UprobesUnwindingVisitor::Visit(uint64_t event_timestamp,
   sample.set_timestamp_ns(event_timestamp);
 
   Callstack* callstack = sample.mutable_callstack();
+  callstack->set_type(Callstack::kComplete);
+
+  bool has_unwinding_error = false;
```

Comment:
I'm not sure if introducing this variable is the best way of dealing with the situation. The function is already quite long and this makes the control flow more complex. What about moving most of the logic below into a separate function and keep the return-early pattern there, and then call this function from here?

(But probably I am missing something.)

Reply:
Extracted a method. PTAL.